File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result=float('-inf')
        curr=0
        i=0
        while i<len(nums):
            curr+=nums[i]
            if curr>result:
                result=curr
            if curr<0:
                curr=0
            i+=1
        return result        

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0,ans=INT_MIN;
        for(int num:nums){
            sum+=num;
            ans=max(ans,sum);
            if(sum<0) sum=0;
        }
        return ans;
    }
};